src/Features/Core/Diagnostics/DiagnosticData.cs

```diff
@@ -284,6 +284,8 @@ private static void SwapIfNeeded(ref LinePosition startLinePosition, ref LinePos
 
         public static DiagnosticData Create(Workspace workspace, Diagnostic diagnostic)
         {
+            Contract.Requires(diagnostic.Location == null || !diagnostic.Location.IsInSource);
```

Comment:
actually, I can't skip checking null since Diagnostic is not sealed. user can return custom diagnostic and Location might return null.